tgui/packages/tgui/interfaces/GasCanister/Detonator.js

```diff
@@ -0,0 +1,165 @@
+import { Fragment } from 'inferno';
+import { Button, LabeledList, Section, Box, Flex, Divider } from '../../components';
+import { DetonatorTimer } from './DetonatorTimer';
+
+export const Detonator = props => {
+  const {
+    detonator,
+    detonatorAttachments,
+    onToggleAnchor,
+    onToggleSafety,
+    onWireInteract,
+    onPrimeDetonator,
+    onTriggerActivate,
+    onSetTimer,
+  } = props;
+
+  return (
+    <Section
+      title="Detonator">
+      <DetonatorWires
+        detonator={detonator}
+        onWireInteract={onWireInteract}
+        onSetTimer={onSetTimer} />
+      <Divider />
+      <DetonatorUtility
+        detonator={detonator}
+        detonatorAttachments={detonatorAttachments}
+        onToggleAnchor={onToggleAnchor}
+        onToggleSafety={onToggleSafety}
+        onPrimeDetonator={onPrimeDetonator}
+        onTriggerActivate={onTriggerActivate} />
+    </Section>
+  );
+};
+
+const DetonatorWires = props => {
+  const {
+    detonator: {
+      wireNames,
+      wireStatus,
+      time,
+      isPrimed,
+    },
+    onWireInteract,
+    onSetTimer,
+  } = props;
+
+  return (
+    <Flex>
+      <Flex.Item>
+        <LabeledList>
+          { wireNames.map((entry, i) => (
+            <LabeledList.Item
+              key={entry + i}
+              label={entry}>
+              { (wireStatus && wireStatus[i]) ? (
+                <Fragment>
+                  <Button
+                    onClick={() => onWireInteract("cut", i)}>
+                    Cut
+                  </Button>
+                  <Button
+                    onClick={() => onWireInteract("pulse", i)}>
+                    Pulse
+                  </Button>
+                </Fragment>)
+                : <Box color="average">Cut</Box> }
+            </LabeledList.Item>
+          ))}
+        </LabeledList>
+      </Flex.Item>
+      <Flex.Item
+        mr={5}
+        mt={2} >
+        <DetonatorTimer
+          time={time} />
+        {!(isPrimed) && (
+          <Button
+            mt={1}
+            onClick={() => onSetTimer()}>
+            Set Timer
+          </Button>)}
```

Comment:
I think I'd be more a fan of this disabling when primed, instead of disappearing entirely

Reply:
Agreed